osi_bank_reconciled_report/models/account_bank_reconciled_report.py

```diff
@@ -252,235 +262,180 @@ def _get_amount(self, line, report_data):
+                statements_lines |= report_data.get('not_reconciled_st_positive')
+            if report_data.get('not_reconciled_st_negative'):
+                statements_lines |= report_data.get('not_reconciled_st_negative')
+            for bank_statement in statements_lines.mapped('statement_id'):
+                lines.append(self._add_bank_statement(bank_statement))
+                for line in report_data.get('not_reconciled_st_positive', []).filtered(lambda l:l.statement_id == bank_statement):
+                    lines.append(self._add_bank_statement_line(line, line.amount, parent_id=bank_statement.id))
+                for line in report_data.get('not_reconciled_st_negative', []).filtered(lambda l:l.statement_id == bank_statement):
+                    lines.append(self._add_bank_statement_line(line, line.amount, parent_id=bank_statement.id))
+
+        if report_data.get('reconciled_st_positive') or \
+                report_data.get('reconciled_st_negative'):
+            lines.append(self.
+                         _add_title_line(options,
+                                         _("Reconciled Bank \
+                                           Statement Lines"),
+                                         level=2))
+            statements_lines = self.env["account.bank.statement.line"]
+            if report_data.get('reconciled_st_positive'):
+                statements_lines |= report_data.get('reconciled_st_positive')
+            if report_data.get('reconciled_st_negative'):
+                statements_lines |= report_data.get('reconciled_st_negative')
+            for bank_statement in statements_lines.mapped('statement_id'):
+                lines.append(self._add_bank_statement(bank_statement))
+                for line in report_data.get('reconciled_st_positive', []).filtered(lambda l:l.statement_id == bank_statement):
+                    lines.append(self._add_bank_statement_line(line, line.amount,parent_id=bank_statement.id))
+    
+                for line in report_data.get('reconciled_st_negative', []).filtered(lambda l:l.statement_id == bank_statement):
+                    lines.append(self._add_bank_statement_line(line, line.amount,parent_id=bank_statement.id))
+
+        if report_data.get('not_reconciled_pmts'):
+            lines.append(self.
+                         _add_title_line(options,
+                                         _("Validated Payments \
                                             Not Linked with a Bank \
-                                            Statement Line"
-                    ),
-                    level=2,
-                )
-            )
-            for line in report_data["not_reconciled_pmts"]:
-                line_number = 0
-                line_number += 1
+                                            Statement Line"), level=2))
+            for line in report_data['not_reconciled_pmts']:
+                self.line_number += 1
```

Comment:
@Vimal-OSI  The mix of self.line_number and the global var line_number is causing an access error, Odoo thinks self.line_number is a readonly field

Reply:
@RLeeOSI , I have checked but I haven't faced any access error with this code.
But technically, self.line_number is accessing the global variable for this class.

Can you please share the reproducible steps ?

Thank you.

Reply:
We have been unable to recreate the error, but PVM sees it every time. That might be because the QA/test data is different, and on prod we're using admin user, who may have more rights. Kali sent the traceback:
```
Odoo Server Error
Traceback (most recent call last):
File "/opt/odoo/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
result = request.dispatch()
File "/opt/odoo/odoo/odoo/http.py", line 684, in dispatch
result = self._call_function(**self.params)
File "/opt/odoo/odoo/odoo/http.py", line 360, in _call_function
return checked_call(self.db, *args, **kwargs)
File "/opt/odoo/odoo/odoo/service/model.py", line 94, in wrapper
return f(dbname, *args, **kwargs)
File "/opt/odoo/odoo/odoo/http.py", line 348, in checked_call
result = self.endpoint(*a, **kw)
File "/opt/odoo/odoo/odoo/http.py", line 913, in __call__
return self.method(*args, **kw)
File "/opt/odoo/odoo/odoo/http.py", line 532, in response_wrap
response = f(*args, **kw)
File "/opt/odoo/odoo/addons/web/controllers/main.py", line 1389, in call_kw
return self._call_kw(model, method, args, kwargs)
File "/opt/odoo/odoo/addons/web/controllers/main.py", line 1381, in _call_kw
return call_kw(request.env[model], method, args, kwargs)
File "/opt/odoo/odoo/odoo/api.py", line 396, in call_kw
result = _call_kw_multi(method, model, args, kwargs)
File "/opt/odoo/odoo/odoo/api.py", line 383, in _call_kw_multi
result = method(recs, *args, **kwargs)
File "/opt/odoo/src/enterprise/account_reports/models/account_report.py", line 1162, in get_report_informations
'main_html': self.get_html(options),
File "/opt/odoo/src/enterprise/account_reports/models/account_report.py", line 1200, in get_html
headers, lines = self._get_table(options)
File "/opt/odoo/src/enterprise/account_reports/models/account_report.py", line 757, in _get_table
return self.get_header(options), self._get_lines(options)
File "/opt/odoo/src/osi-addons/osi_bank_reconciled_report/models/account_bank_reconciled_report.py", line 351, in _get_lines
self.line_number += 1
Exception
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/opt/odoo/odoo/odoo/http.py", line 640, in _handle_exception
return super(JsonRequest, self)._handle_exception(exception)
File "/opt/odoo/odoo/odoo/http.py", line 316, in _handle_exception
raise exception.with_traceback(None) from new_cause
AttributeError: 'account.bank.reconciled.report' object attribute 'line_number' is read-only
```
The steps to reproduce are: In SC company, open Accounting app, in Kanban view of Journals click 3 dots on the Chase journal, select Reconciled report (NOT reconciliation report)

I agree that this doesn't really make sense, and self.line_number should be calling the global var, but based on the previous code omitting the self. and from the error returned, let's try with line_number alone and see if that clears the error